packages/cli/src/base.ts

```diff
@@ -201,28 +194,27 @@ export abstract class BaseCommand extends Command {
       kit.defaultAccount = res.flags.from
     }
 
-    const gasCurrencyConfig = res.flags.gasCurrency
-      ? (gasOptions as any)[res.flags.gasCurrency]
-      : getGasCurrency(this.config.configDir)
+    const gasCurrencyFlag = res.flags.gasCurrency ?? getGasCurrency(this.config.configDir)
+    if (gasCurrencyFlag) {
+      const feeCurrencyWhitelist = await kit.contracts.getFeeCurrencyWhitelist()
+      const validFeeCurrencies = await feeCurrencyWhitelist.getWhitelist()
+      if (validFeeCurrencies.includes(gasCurrencyFlag)) {
```

Comment:
do we maybe want to make it case insensitive? right now afaik it is:

```js
> ["0xAddress"].includes("0xaddress")
false

> ["0xAddress"].includes("0xAddress")
true
```

Reply:
and from what I checked it is actually insensitive when passing it to the client

Reply:
good idea.

Reply:
not sure what value it brings as we're comparing addresses stored in a contract to user-input, case-insenstive comparison makes more sense IMHO